Consul.Test/BaseFixture.cs

```diff
@@ -0,0 +1,50 @@
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Consul.Test
+{
+    public class BaseFixture : IAsyncLifetime
+    {
+        protected ConsulClient _client;
+
+        public BaseFixture()
+        {
+            _client = new ConsulClient(c =>
+            {
+                c.Token = TestHelper.MasterToken;
+                c.Address = TestHelper.HttpUri;
+            });
+        }
+
+        public Task DisposeAsync()
+        {
+            _client.Dispose();
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// With parallel testing on the CI, sometimes the tests start before the consul server finished properly initializing
+        /// This aims to workaround it in a not so elegant way. https://github.com/hashicorp/consul/issues/819
+        /// </summary>
+        /// <returns></returns>
+        public async Task InitializeAsync()
+        {
+            bool ready = false;
+
+            while (!ready)
```

Comment:
I think we need a timeout, otherwise it will wait forever for the consul agent when it is not started.

Reply:
The risk of consul server not starting is rather low I'd say. On the rare chance that it may occur, the CI build will just fail after the GitHub Action times out so not a problem IMO. Let me know.

Reply:
I'm more concerned about local testing, devs can get very confused when they forget to start Consul before running tests.

Reply:
Done